### ISSUE TITLE
The `to_windows` method yields an odd error and doesn't generate the windows.

### ISSUE THREAD

**Issue by phildias** [tariqshihadah/linref#4]
Here's the sample data I'm using:  https://www.dropbox.com/scl/fi/r3zpba1gknl3d0hvxencz/linref_bug.zip?rlkey=d4y1lcun3urm07f6pb6w4m9ki&dl=0

For some reason, I can't get the `to_windows` method to work with this particular input data. 
I'm not sure what's so special about this GeoDataFrame, but when I run the `to_windows` method on it, I get an error message that sounds like its trying to interpret the `beg` column as a string instead of a double. 

```python
import geopandas as gpd
import linref as lr

# Input layer is in EPSG:3081
srcc_missing = gpd.read_file('c:/temp/linref_bug.gpkg')

# Making the Events Collection object
srcc_missing_ec = lr.EventsCollection(srcc_missing.copy(), 
                                      keys=['RIA_RTE_ID'], 
                                      beg='START', 
                                      end='END',
                                      geom='geometry')

# Trying to make the new non-overlapping windows. 
srcc_windows_ec = srcc_missing_ec.to_windows(length=100, 
                                             steps=1, 
                                             fill='balance')
```
The code above yields the following error traceback:
```
Traceback (most recent call last):

  File "C:\Users\diasf\AppData\Local\Temp\ipykernel_18048\1850216886.py", line 1, in <cell line: 1>
    srcc_windows_ec = srcc_missing_ec.to_windows(length=100,

  File "C:\ProgramData\Anaconda3\envs\gpd\lib\site-packages\linref\events\collection.py", line 1146, in to_windows
    df = df.astype(dtypes, copy=False)

  File "C:\ProgramData\Anaconda3\envs\gpd\lib\site-packages\pandas\core\generic.py", line 6226, in astype
    res_col = col.astype(dtype=cdt, copy=copy, errors=errors)

  File "C:\ProgramData\Anaconda3\envs\gpd\lib\site-packages\pandas\core\generic.py", line 6240, in astype
    new_data = self._mgr.astype(dtype=dtype, copy=copy, errors=errors)

  File "C:\ProgramData\Anaconda3\envs\gpd\lib\site-packages\pandas\core\internals\managers.py", line 448, in astype
    return self.apply("astype", dtype=dtype, copy=copy, errors=errors)

  File "C:\ProgramData\Anaconda3\envs\gpd\lib\site-packages\pandas\core\internals\managers.py", line 352, in apply
    applied = getattr(b, f)(**kwargs)

  File "C:\ProgramData\Anaconda3\envs\gpd\lib\site-packages\pandas\core\internals\blocks.py", line 526, in astype
    new_values = astype_array_safe(values, dtype, copy=copy, errors=errors)

  File "C:\ProgramData\Anaconda3\envs\gpd\lib\site-packages\pandas\core\dtypes\astype.py", line 299, in astype_array_safe
    new_values = astype_array(values, dtype, copy=copy)

  File "C:\ProgramData\Anaconda3\envs\gpd\lib\site-packages\pandas\core\dtypes\astype.py", line 230, in astype_array
    values = astype_nansafe(values, dtype, copy=copy)

  File "C:\ProgramData\Anaconda3\envs\gpd\lib\site-packages\pandas\core\dtypes\astype.py", line 170, in astype_nansafe
    return arr.astype(dtype, copy=True)

ValueError: invalid literal for int() with base 10: '0.0'
```

The `to_windows` method seems to work perfectly well for another dataset, but something about this is tripping it up. Am I missing something just super simple and obvious? 

Thanks!

**Comment by tariqshihadah** [comment]
@phildias Thanks for bringing this up, I'll take a look!

Could you confirm what version of `linref` you're using?

Could you provide the output of `srcc_missing_ec.info()`?

Edit: No longer needed, issue should be solved. Was able to replicate the error and solve it pretty quickly.

**Comment by tariqshihadah** [comment]
Opened a [new branch](https://github.com/tariqshihadah/linref/tree/windows_dtype_issue) to address this item. @phildias please review the [PR](https://github.com/tariqshihadah/linref/pull/5) and ensure that this fixes your issue. If it does, please approve and I'll merge.

**Comment by phildias** [comment]
Yep, that did the trick! Thanks for the quick fix!!!

**Comment by tariqshihadah** [comment]
Issue solved with PR #5.
